src/context/jobListingContext.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { createContext, useContext, useReducer } from 'react';
+import jobDataArr from '../data/data.json';
+// import { JobTransformed } from '../types/JobListing';
+import { transformJobListing } from '../utilities';
+import type { JobData, SearchFilterAction} from '../types/JobListing';
+import {jobListingReducer} from '../reducers/jobListingReducer';
+const jobDataTransformed = transformJobListing(jobDataArr);
+
+const defaultValues: JobData = {
+  jobData: jobDataTransformed,
+  searchFilters: [],
+  filteredJobListings: []
+};
+
+const jobContext = {
+  jobData: defaultValues,
+  setJobData: (action: SearchFilterAction): void => {}
+};
+
+const JobListingContext =
+  createContext<{
+    jobData: JobData;
+    setJobData: React.Dispatch<SearchFilterAction>;
+  }>(jobContext);
+
+interface Props {
+  children: React.ReactNode;
+}
+
+export const JobListingProvider: React.FC<Props> = ({ children }) => {
+  const [jobData, setJobData] = useReducer(jobListingReducer, defaultValues);
```

Comment:
This is another common trap, destructuring is a _convenience_, it's not _required_!

There are different solutions:

- not using destructuring when it would lead to names colliding or confusion

    ```ts
    const jobListings = useJobListing();
    const foo = useFoo();

    jobListings.dispatch(/**/);
    foo.dispatch(/**/);
    ```
- aliasing names locally, as you mentioned

    ```ts
    const {dispatch: dispatchListingEvent} = useJobListing();
    const {dispatch: dispatchFooEvent} = useFoo();

    dispatchListingEvent(/**/);
    dispatchFooEvent(/**/);
    ```

Personally, I prefer falling back to _not_ destructuring when it just leads to aliasing all the props with either a prefix or a suffix (e.g. `prefixProperty` or `propertySuffix`), which in the end is similar to the dot notation anyway (e.g. `prefix.property`) but without all the noise of aliasing everytime 🎉